Fix for DOC-450, 451, 452, 453 (#268)
Macrometacorp/docs: docs/cep/reference/functions/adhoc/adhoc.md

---

> title: Adhoc Query

Reviewer: ```suggestion
title: Ad Hoc Queries
```

---

> Adhoc queries provide a way of performing ad-hoc operations on tables (stores), named-windows, and named-aggregations. We can send adhoc queries and fetch data from Stores and named windows.

Reviewer: ```suggestion
Ad hoc queries provide a way of performing ad-hoc operations on tables (stores), named windows, and named aggregations. We can send ad hoc queries and fetch data from stores and named windows.
```

Reviewer: Can you give a little more explanation of what an ad hoc query is in this context? Saying an ad hoc query lets me perform "ad hoc operations" doesn't tell me anything if I don't know what ad hoc either thing is.

> 

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

## Syntax

> Adhoc Query uses the following syntax:

Reviewer: ```suggestion
Ad hoc queries use the following syntax:
```

```js
	select * from SampleAdhocQueryInputTableOneMinTimeWindow;
	
	SELECT * FROM SampleAdhocQueryTable;
```				  

> 					  

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

## Example

> 

Reviewer: Please add an intro with some context of what is in this example.

```js
-- Defines `SampleAdhocQueryInputTable` collection to process events having `sensorId` and `temperature`(F).
CREATE SOURCE SampleAdhocQueryInputTable WITH(type = 'database', collection = "SampleAdhocQueryInputTable", collection.type="doc" , replication.type="global", map.type='json') (sensorId string, temperature double);

-- Named Window
CREATE WINDOW SampleAdhocQueryInputTableOneMinTimeWindow (sensorId string, temperature double) SLIDING_TIME(1 min);

-- Table
CREATE TABLE SampleAdhocQuerySensorA1234DestTable(sensorId string, temperature double);

@info(name = 'Insert-to-window')
INSERT INTO SampleAdhocQueryInputTableOneMinTimeWindow
SELECT *
FROM SampleAdhocQueryInputTable;

@info(name = 'EqualsFilter')

> -- Note: Filter out events with `sensorId` equalling `sensor A1234`

Reviewer: ```suggestion

-- Note: Filter out events with `sensorId` equalling `sensor A1234`
```

INSERT INTO SampleAdhocQuerySensorA1234DestTable
SELECT *
FROM SampleAdhocQueryInputTable
WHERE sensorId == 'sensor A1234';
```

> 1. Upload following data into `SampleAdhocQueryInputTable` C8DB Collection

Reviewer: What are these? Do the numbers correspond to parts of the code block above? I am very confused.

Reviewer: Also, where do I do any of this? Query editor? Stream worker screen?

	{"sensorId":"sensor A1234","temperature":18}
	{"sensorId":"sensor A1234","temperature":-32.2}
	{"sensorId":"sensor FR45","temperature":20.9}
	{"sensorId":"sensor meter1","temperature":49.6}

2. This application accumulates all the data for one minute in the named window `SampleAdhocQueryInputTableOneMinTimeWindow`
	Named window allows other application to query data in realtime.

> 3. Run the adhoc query on the `SampleAdhocQueryInputTableOneMinTimeWindow` 

Reviewer: ```suggestion
3. Run the ad hoc query on the `SampleAdhocQueryInputTableOneMinTimeWindow`.
```
